yb-voyager/src/query/queryissue/detectors_ddl.go

```diff
@@ -963,6 +984,25 @@ func (aid *AlterTableIssueDetector) DetectIssues(obj queryparser.DDLObject) ([]Q
 			}
 
 		}
+
+		if alter.ConstraintType == queryparser.PRIMARY_CONSTR_TYPE {
+			//Report PRIMARY KEY (createdat timestamp) as hotspot issue
+			if len(alter.ConstraintColumns) > 0 {
```

Comment:
logic seems identical to the PK logic. Can you take it out into a function that takes the required inputs (constraintColumns, columnsWithHotspotRangeIndexesDatatypes, etc) ?